argocd/structure_cluster.go

```diff
@@ -177,15 +177,8 @@ func flattenClusterInfo(info application.ClusterInfo) []map[string]interface{} {
 func flattenClusterConfig(config application.ClusterConfig, d *schema.ResourceData) []map[string]interface{} {
 	r := map[string]interface{}{
 		"username":             config.Username,
-		"exec_provider_config": flattenClusterConfigExecProviderConfig(config.ExecProviderConfig),
-	}
-
-	if stateClusterConfig, ok := d.GetOk("config"); ok {
-		scc := expandClusterConfig(stateClusterConfig.([]interface{})[0])
```

Comment:
Rather than expanding the existing state as was done previously, this change makes use of the `TLSClientConfig` that is returned by the API, which ensures that we track the remote state of the attributes the API does return (I.e. `CAData`, `CertData`, `Insecure` and `ServerName`).